lib/matplotlib/quiver.py

```diff
@@ -114,26 +122,56 @@
 scale : float, optional
     Scales the length of the arrow inversely.
 
-    Number of data units per arrow length unit, e.g., m/s per plot width; a
-    smaller scale parameter makes the arrow longer. Default is *None*.
-
-    If *None*, a simple autoscaling algorithm is used, based on the average
-    vector length and the number of vectors. The arrow length unit is given by
-    the *scale_units* parameter.
+    Number of data values represented by one unit of arrow length on the plot.
+    For example, if the data represents velocity in meters per second (m/s), the
+    scale parameter determines how many meters per second correspond to one unit of
+    arrow length relative to the width of the plot.
+    Smaller scale parameter makes the arrow longer.
 
-scale_units : {'width', 'height', 'dots', 'inches', 'x', 'y', 'xy'}, optional
-    If the *scale* kwarg is *None*, the arrow length unit. Default is *None*.
+    If *None*, a autoscaling algorithm is used to scale the arrow lenght to a
+    reasonable size, which is based on the average vector length and the number of
+    vectors.If the autoscaling does not yield a reasonable result, the user has to
+    manually define a scale.
 
-    e.g. *scale_units* is 'inches', *scale* is 2.0, and ``(u, v) = (1, 0)``,
-    then the vector will be 0.5 inches long.
+    The arrow length unit is given by the *scale_units* parameter.
 
-    If *scale_units* is 'width' or 'height', then the vector will be half the
-    width/height of the axes.
-
-    If *scale_units* is 'x' then the vector will be 0.5 x-axis
-    units. To plot vectors in the x-y plane, with u and v having
-    the same units as x and y, use
-    ``angles='xy', scale_units='xy', scale=1``.
+scale_units : {'width', 'height', 'dots', 'inches', 'x', 'y', 'xy'}, optional
+    If scale_units is None, the length of the arrows is effectively same as when it
+    is set to 'width', meaning the arrow length will be relative to the width of
+    the axes.
+
```

Comment:
We should give a one-sentence explanation what this is. I also believe we can generalize and pull out the formula:

```suggestion
    The physical image unit, which is used for rendering the scaled arrow data *U*, *V*.

    The rendered arrow length is given by
    
        length in x direction = $\frac{u}{\mathrm{scale}} \mathrm{scale_unit}$
    
        length in y direction = $\frac{u}{\mathrm{scale}} \mathrm{scale_unit}$
        
   For example, ``(u, v) = (0.5, 0)`` with ``scale=10, scale_unit="width"`` results
   in a horizontal arrow with a length of *0.5 / 10 * "width"*, i.e. 0.05 times the
   Axes width.
   
   Supported values are:
    
```